pairings.py

```diff
@@ -0,0 +1,16 @@
+import random
+
+
+#assigned pairings
+pairs = []
+
+def pair_random(players): 
+    if len(players)%2 != 0: players.append('None')
+    random.shuffle(players)
+
+    #create pairing tuples and add to pairings list
+    while len(players) > 1 :  
+        pairs.append((players[0], players [-1]))
+        players.pop(-1)
+        players.pop(0)

```

Comment:
First of all, you shouldn't be indexing pop. It just takes an element from the end of a list. You're taking different element than expected here.

Secondly, pop returns the element it took. You can use it to construct the pair.

Thirdly, you can use comprehension to construct the list. Consider:
```suggestion
   pairs = [ (c.pop(), c.pop()) for _ in range(len(c)//2) ]
   if c: pairs.append((c.pop(), None))
```

Reply:
I amended the code to this:
`import random

def pair_random(players): 
    
    random.shuffle(players)
   
    pairs = [ (players.pop(), players.pop()) for _ in range(len(players)//2) ]
    if len(players)%2 != 0: pairs.append((players.pop(),None))`

it's returning none, have i done something wrong?

Reply:
Fixed it, forgot to return pairs.
It works